src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,125 @@
+        private bool TryClassifyCallArgument(SyntaxNode syntax, SemanticModel semanticModel, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (!(syntax is ArgumentSyntax argument))
+            {
+                return false;
+            }
+
+            // in arguments we don't need to go on if the out is missing or the argument isn't inside a tuple
+            if (!argument.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword) &&
+                !argument.IsParentKind(SyntaxKind.TupleExpression))
+            {
+                return false;
+            }
+
+            var operation = semanticModel.GetOperation(argument.Expression, cancellationToken);
+            if (!(operation is IDiscardOperation discardOperation))
+            {
+                return false;
+            }
+
+            switch (discardOperation.Syntax)
+            {
+                case DeclarationExpressionSyntax decl:
+                    result.Add(new ClassifiedSpan(decl.Designation.Span, ClassificationTypeNames.Keyword));
+                    return true;
+                case IdentifierNameSyntax discard:
+                    result.Add(new ClassifiedSpan(discard.Span, ClassificationTypeNames.Keyword));
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        private bool TryClassifyLambdaParameter(SyntaxNode syntax, SemanticModel semanticModel, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (!(syntax is LambdaExpressionSyntax lambda))
+            {
+                return false;
+            }
+
+            var symbolInfo = semanticModel.GetSymbolInfo(lambda, cancellationToken);
+            var symbol = symbolInfo.Symbol as IMethodSymbol;
+
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            DataFlowAnalysis? _lazyAnalysis = null;
+
+            foreach (var parameter in symbol.Parameters)
+            {
+                if (parameter.IsDiscard)
+                {
+                    result.Add(new ClassifiedSpan(parameter.Locations[0].SourceSpan, ClassificationTypeNames.Keyword));
+                }
+                else if (parameter.Name == "_") //unfortunately parameter.IsDiscard currently doesn't work for all cases
+                {
+                    if (_lazyAnalysis == null)
+                    {
+                        _lazyAnalysis = semanticModel.AnalyzeDataFlow(lambda.Body);
+                    }
+
+                    //make sure the parameter isn't used in the lambda, in which case it wouldn't be discarding
+                    if (!_lazyAnalysis.ReadInside.Contains(parameter))
+                    {
+                        result.Add(new ClassifiedSpan(parameter.Locations[0].SourceSpan, ClassificationTypeNames.Keyword));
+                    }
+                }
+            }
+
+            return true;
```

Comment:
seems weird to return 'true' here given that it may not have added anything.

Reply:
got rid of it